src/cpp/fastdds/domain/DomainParticipantImpl.cpp

```diff
@@ -349,21 +356,21 @@ ReturnCode_t DomainParticipantImpl::enable()
 
             for (auto sub : subscribers_)
             {
-                sub.second->rtps_participant_ = rtps_participant_;
+                sub.second->rtps_participant_ = get_rtps_participant();
                 sub.second->user_subscriber_->enable();
             }
         }
     }
 
-    rtps_participant_->enable();
+    get_rtps_participant()->enable();
 
     return ReturnCode_t::RETCODE_OK;
 }
 
 ReturnCode_t DomainParticipantImpl::set_qos(
         const DomainParticipantQos& qos)
 {
-    bool enabled = (rtps_participant_ != nullptr);
+    bool enabled = (get_rtps_participant() != nullptr);
```

Comment:
Call `get_rtps_participant()` once into local variable and use it later ?

Perhaps even taking `mtx_gs_` and using `rtps_participant_` directly ?

Reply:
Solved by taking `mtx_gs_` for most of the method, except the call to `update_attributes`